DragonLoop/DragonLoopApp/DragonLoopApp/ViewModels/MapViewModel.cs

```diff
@@ -35,13 +38,54 @@ public MapViewModel() : base(Settings.UrlBase)
                         };
         }
 
-        private async Task ExecuteLoadRoutesCommand()
+        private async Task ExecuteLoadDataCommand()
         {
             if (IsBusy)
                 return;
 
             IsBusy = true;
 
+            try
+            {
+                await ExecuteLoadBuses();
+                await ExecuteLoadRoutes();
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex);
+            }
+            finally
+            {
+                IsBusy = false;
+            }
+        }
+
+        public async Task ExecuteLoadBuses()
+        {
+            try
+            {
+                BusCollection.Clear();
+                await LoadBuses();
+                foreach (var bus in Buses)
+                {
+                    BusCollection.Add(bus);
+                    var pin = new Pin
```

Comment:
I feel like we should tie adding the Bus pins to the route sliders just like you did for the stops right? It seems weird to show all the buses all the time and could get hard to see what the user actually cares about.

Reply:
ok i can do that

Reply:
This would require rewrite of all the code i have but i will be doing this